src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -231,17 +231,26 @@ vec3 irr_glsl_diffuseFresnelCorrectionFactor(in vec3 n, in vec3 n2)
 #define _IRR_BSDF_BRDF_SPECULAR_BLINN_PHONG_INCLUDED_
 
 #include <irr/builtin/glsl/bsdf/common.glsl>
+#include <irr/builtin/glsl/bsdf/brdf/specular/fresnel/fresnel.glsl>
 
 float irr_glsl_blinn_phong(in float NdotH, in float n)
 {
-    return pow(NdotH, n);
+    float nom = n*(n + 6.0) + 8.0;
+    float denom = pow(0.5, 0.5*n) + n;
+    float normalization = 0.125 * irr_glsl_RECIPROCAL_PI * nom/denom;
+    return normalization*pow(NdotH, n);
 }
 
-float irr_glsl_blinn_phong_normalizationFactor(in float n)
+float irr_glsl_blinn_phong_fresnel_dielectric_cos_eval(in irr_glsl_BSDFIsotropicParams params, in float n, in float ior)
```

Comment:
vec3 ior?

Reply:
fresnel_dielectric takes ior as a single value

Reply:
ok, but there's really no reaon for it to not support a per-RGB channel ior value

Reply:
yea, well im not sure why lagarde made it like this, i'm making it vec3

Reply:
because he optimized to extremes